### PR TITLE
read hosts from env vars

### PR DESCRIPTION
**Pull request by dickeyxxx**

### CODE REVIEW

**Comment by tt** [comment]
You need [this line](https://github.com/heroku/heroku-cli-util/blob/73916a8d8640274a8b67bf9444fb7cdd261b8137/lib/auth.js#L113) too.

File: lib/vars.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const url = require('url')
+
+class Vars {
+  constructor (env) {
+    this.env = env
+  }
+
+  get host () { return this.env.HEROKU_HOST || 'heroku.com' }
+  get apiHost () { return this.host.startsWith('http') ? this.host : `api.${this.host}` }
```
Comment (tt) [inline lib/vars.js]: Why can it start with "http"? I don't think that will work with the .netrc format when it tries to fetch the token.

**Comment by dickeyxxx** [comment]
previous code is here for reference: https://github.com/heroku/cli/blob/master/git.go

File: lib/vars.js (hunk 1 above)
Comment (KarateCowboy) [inline lib/vars.js]: That looks like a good call for a unit test.

File: lib/vars.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  get httpGitHost () {
+    if (this.env.HEROKU_GIT_HOST) return this.env.HEROKU_GIT_HOST
+    if (this.host.startsWith('http')) {
+      const u = url.parse(this.host)
+      if (u.host) return u.host
+    }
+    return `git.${this.host}`
+  }
+
+  get gitPrefixes () {
```
Comment (KarateCowboy) [inline lib/vars.js]: This doesn't seem to be used anywhere. Is it?

Comment (dickeyxxx) [inline lib/vars.js]: yeah I suppose we don't need to include everything

Comment (dickeyxxx) [inline lib/vars.js]: ok, removed

File: lib/vars.js (hunk 1 above)
Comment (dickeyxxx) [inline lib/vars.js]: should be one now

File: lib/auth.js [hunk 3]
```diff
@@ -33,10 +34,10 @@ function createOAuthToken (username, password, secondFactor) {
 function saveToken ({email, token}) {
   const Netrc = require('netrc-parser')
   const netrc = new Netrc()
-  const hosts = ['api.heroku.com', 'git.heroku.com']
+  const hosts = [vars.apiHost, vars.gitHost]
   hosts.forEach(host => {
-    netrc.machines[host].login = email
-    netrc.machines[host].password = token
+    netrc.machines[vars.apiHost].login = email
```
Comment (tt) [inline lib/auth.js]: Now you're always setting the `vars.apiHost` and not the actual `host` which will break all Git clients...

**Pull request merged**